[DCGM] Installation feedback (#15355)
DataDog/integrations-core: dcgm/README.md

### Installation

> The DCGM check is included in the [Datadog Agent][1] package. However, you need to spin up the DCGM Exporter container to expose the GPU metrics in order for the Agent to collect this data. Datadog recommends using the following DCGM configuration to cover the same ground as the NVML integration as well as additional useful metrics, as the default counters are not sufficient.

Reviewer: ```suggestion
The DCGM check is included in the [Datadog Agent][1] package. However, you need to spin up the DCGM Exporter container to expose the GPU metrics in order for the Agent to collect this data. As the default counters are not sufficient, Datadog recommends using the following DCGM configuration to cover the same ground as the NVML integration in addition to having useful metrics.
```

<!-- xxz tab xxx -->
<!-- xxx tab "Kubernetes" xxx -->

#### Kubernetes (DCGM exporter Helm chart)

> The DCGM exporter can quickly be installed in a Kubernetes environment using NVIDIA DCGM Exporter Helm chart. The instructions below are derived from the template provided by NVIDIA [here](https://github.com/NVIDIA/dcgm-exporter#quickstart-on-kubernetes).

Reviewer: ```suggestion
The DCGM exporter can quickly be installed in a Kubernetes environment using the NVIDIA DCGM Exporter Helm chart. The instructions below are derived from the template provided by NVIDIA [here](https://github.com/NVIDIA/dcgm-exporter#quickstart-on-kubernetes).
```

1. Add the NVIDIA DCGM Exporter Helm repository and ensure it is up-to-date :
```bash
helm repo add gpu-helm-charts https://nvidia.github.io/dcgm-exporter/helm-charts && helm repo update
```

> 2. Create a `ConfigMap` holding the Datadog-recommended metrics from [Installation](#Installation), as well as the `RoleBinding` and `Role` used by the DCGM pods to retrieve the `ConfigMap` using the manifest below :

Reviewer: ```suggestion
2. Create a `ConfigMap` containing the Datadog-recommended metrics from [Installation](#Installation), as well as the `RoleBinding` and `Role` used by the DCGM pods to retrieve the `ConfigMap` using the manifest below :
```

    {
      "dcgm": {
        "instances": [
          {
            "openmetrics_endpoint": "http://%%host%%:9400/metrics"
          }
        ]
      }
    }

# Optional - Disabling the ServiceMonitor which requires Prometheus CRD - can be re-enabled if Prometheus CRDs are installed in your cluster
serviceMonitor:
  enabled: false
```
4. Install the DCGM Exporter Helm chart in the `default` namespace with the following command, while being in the directory with your `dcgm-values.yaml` :
```bash
helm install dcgm-datadog gpu-helm-charts/dcgm-exporter -n default -f dcgm-values.yaml
```

**Note**: You can modify the release name `dcgm-datadog` as well as the namespace, but you must modify accordingly the manifest from step 1.

<!-- xxz tab xxx -->
<!-- xxx tab "Operator" xxx -->

#### Kubernetes (NVIDIA GPU Operator)

The DCGM exporter can be installed in a Kubernetes environment by using NVIDIA GPU Operator. The instructions below are derived from the template provided by NVIDIA [here](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html).

1. Add the NVIDIA GPU Operator Helm repository and ensure it is up-to-date :

> ```bash

Reviewer: ```suggestion
   ```bash
```

> helm repo add nvidia https://helm.ngc.nvidia.com/nvidia && helm repo update

Reviewer: ```suggestion
    helm repo add nvidia https://helm.ngc.nvidia.com/nvidia && helm repo update
```

> ```

Reviewer: ```suggestion
   ```
```

Author: @rtrieu thanks for all the feedback/changes, just not understanding this one, we want to close this block of code ?

Reviewer: hi @tbavelier sorry I don't know why it's suggesting to remove it - what I meant to say was to indent it by 3 lines to line it up with the numbered bullet, not to delete it :)

Author: Ah, that makes more sense, thanks a lot! Dismissed your review as I aligned everything now to be inside the number bullet, that's cleaner!

2. Follow the [Custom Metrics Config](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html#custom-metrics-config) instructions with the CSV from [Installation](#installation) :

>     * Fetch the metrics file and save as dcgm-metrics.csv : `curl https://raw.githubusercontent.com/NVIDIA/dcgm-exporter/main/etc/dcp-metrics-included.csv > dcgm-metrics.csv`

Reviewer: ```suggestion
    * Fetch the metrics file and save as `dcgm-metrics.csv`: `curl https://raw.githubusercontent.com/NVIDIA/dcgm-exporter/main/etc/dcp-metrics-included.csv > dcgm-metrics.csv`
```

    * Edit the metrics file by replacing its content with the Datadog-provided mapping.

>     * Create a namespace `gpu-operator` if one is not already present : `kubectl create namespace gpu-operator`.

Reviewer: ```suggestion
    * Create a namespace `gpu-operator` if one is not already present: `kubectl create namespace gpu-operator`.
```

>     * Create a ConfigMap using the file edited above : `kubectl create configmap metrics-config -n gpu-operator --from-file=dcgm-metrics.csv`

Reviewer: ```suggestion
    * Create a ConfigMap using the file edited above: `kubectl create configmap metrics-config -n gpu-operator --from-file=dcgm-metrics.csv`
```

> 3. Create your GPU Operator Helm chart `dcgm-values.yaml` with the following content : 

Reviewer: ```suggestion
3. Create your GPU Operator Helm chart `dcgm-values.yaml` with the following content: 
```

```yaml
# Refer to NVIDIA documentation for the driver and toolkit for your GPU-enabled nodes - example below for Amazon Linux 2 g5.xlarge
driver:
  enabled: true
toolkit:
  version: v1.13.5-centos7
# Using custom metrics configuration to collect recommended Datadog additional metrics - requires the creation of the metrics-config ConfigMap from the previous step
# Ref: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html#custom-metrics-config
dcgmExporter:
  config:
    name: metrics-config
  env:
  - name: DCGM_EXPORTER_COLLECTORS
    value: /etc/dcgm-exporter/dcgm-metrics.csv

# Adding Datadog autodiscovery V2 annotations
daemonsets:
  annotations:
    ad.datadoghq.com/nvidia-dcgm-exporter.checks: |-
      {
        "dcgm": {
          "instances": [
            {
              "openmetrics_endpoint": "http://%%host%%:9400/metrics"
            }
          ]
        }
      }
```

> 4. Install the DCGM Exporter Helm chart in the `default` namespace with the following command, while being in the directory with your `dcgm-values.yaml` :

Reviewer: ```suggestion
4. Install the DCGM Exporter Helm chart in the `default` namespace with the following command, while being in the directory with your `dcgm-values.yaml`:
```
